src/log_surgeon/finite_automata/RegexNFA.hpp

```diff
@@ -118,6 +131,19 @@ auto RegexNFA<NFAStateType>::new_state_with_negative_tagged_transition(
     return m_states.back().get();
 }
 
+template <typename NFAStateType>
+auto RegexNFA<NFAStateType>::new_capture_group_start_states(
+        Tag const* tag,
+        NFAStateType const* dest_state
+) -> std::pair<NFAStateType*, NFAStateType*> {
+    auto* start_state = new_state();
+    m_root->add_positive_tagged_start_transition(tag, start_state);
+
+    auto* end_state = new_state_with_positive_tagged_end_transition(tag, dest_state);
+
+    return {start_state, end_state};
```

Comment:
```suggestion
    auto* end_state = new_state_with_positive_tagged_end_transition(tag, dest_state);
    return {start_state, end_state};
```

Personally I think adding these two empty lines aren't making the code more readable but a little more confusing though...

Reply:
I agree the second line is unneccessary, but I think the first line makes a clear separation b/w the operations pertaining to the `start_state` and the operations pertaining to the `end_state`

Reply:
Imo the code is short enough for people to understand stand without empty lines. Adding empty lines in such a short function looks weird

Reply:
I'm going to disagree here. Our general standard so far has been to use newlines to separate logically distinct sections of the code. We create and handle the start state, then we create the end state.

Reply:
sure, let's keep the empty line between the creation of two states.